packages/nimble-components/src/banner/tests/banner.stories.ts

```diff
@@ -0,0 +1,130 @@
+        <nimble-banner
+            ?open="${x => x.open}"
+            severity="${x => x.severity}"
+            ?title-hidden="${x => x.titleHidden}"
+            ?prevent-dismiss="${x => x.preventDismiss}"
+            dismiss-button-label="Close"
+        >
+            <span slot="title">${x => x.title}</span>
+            ${x => x.text}
+            ${when(x => x.action === 'button (ghost)', html`
+                <nimble-button slot="action" appearance="ghost">Do action</nimble-button>`)}
+            ${when(x => x.action === 'button (outline)', html`
+                <nimble-button slot="action" appearance="outline">Do action</nimble-button>`)}
+            ${when(x => x.action === 'anchor', html`
+                <nimble-anchor slot="action" href="#">Go to site</nimble-anchor>`)}
+        </nimble-banner>
+`),
+    // eslint-disable-next-line storybook/no-redundant-story-name
+    storyName: 'Banner',
+    argTypes: {
+        open: {
+            description: 'Controls whether the banner is visible.'
+        },
+        title: {
+            description:
+                'Primary text which is displayed when `title-hidden` is not set. Banners should **always include a title** even when `title-hidden` is set. The title is used to provide an accessible name to assistive technologies regardless of the value of `title-hidden`.<br><br>Provide the title in an `inline` element such as `<span>` that is targeted to the `title` slot.'
+        },
+        text: {
+            description:
+                'Secondary text that provides a detailed message. This text should be short enough to fit within three lines when displayed.<br><br>Provide this text as content of the banner element (uses the default slot).'
+        },
+        severity: {
+            options: Object.values(BannerSeverity),
+            control: { type: 'radio' },
+            description:
+                'Severity of the message presented by the banner. Controls the displayed color and icon. If not set, the banner has a neutral appearance.'
+        },
+        action: {
+            options: Object.values(ActionType),
+            control: { type: 'radio' },
+            description:
+                'The `action` slot provides a place to display a button or anchor that you provide. If you provide a button, it should have either the `"ghost"` or `"outline"` appearance.'
+        },
+        preventDismiss: {
+            name: 'prevent-dismiss',
+            description:
+                'If set, removes the "X" button that dismisses the banner.'
+        },
+        titleHidden: {
+            name: 'title-hidden',
+            description: 'If set, hides the provided title.'
+        },
+        dismissButtonLabel: {
+            name: 'dismiss-button-label',
+            description:
+                'Set to a localized label (e.g. `"Close"`) for the dismiss button. This provides an accessible name for assistive technologies.'
+        },
+        toggle: {
```

Comment:
I think we really ought to document the events explicitly. It seems weird to just make it a game of "find the events". Our documentation needs a more thorough audit for consistency and completeness, I think. There are holes (like the events) as well as minimal high-level documentation that need to be addressed. I could file an issue, but I'm sure it would just rot, since this is a big task that needs some discussion and probably some coordinated effort. How do you think we should proceed?

Reply:
> Our documentation needs a more thorough audit for consistency and completeness

I had started that discussion about capturing more in storybook and being consistent in standups and came up with a possible approach that aligns with the direction you are going in the PR (document the event explicitly as a separate entry in the controls table): https://github.com/ni/nimble/issues/824

If there is additional things we should capture I'd recommend that issue

Reply:
I'm good with adding this event as you have it here and tracking broader adoption of the pattern in the issue Milan linked.